reco_utils/recommender/deeprec/io/dkn_iterator.py

```diff
@@ -129,6 +142,8 @@ def load_data_from_file(self, infile):
 
         Returns:
             obj: An iterator that will yields parsed results, in the format of graph feed_dict.
+            List: impression id list
+            Int: size of the data in a batch
```

Comment:
I think the codes in this function can be dramatically reduced by using a 2d list containing all the columns. E.g.,
```
candidate_news_index_batch = []
        candidate_news_val_batch = []
        click_news_index_batch = []
        click_news_val_batch = []
        candidate_news_entity_index_batch = []
        click_news_entity_index_batch = []
        label_list = []
        impression_id_list = []

...

(
                    label,
                    candidate_news_index,
                    candidate_news_val,
                    click_news_index,
                    click_news_val,
                    candidate_news_entity_index,
                    click_news_entity_index,
                    impression_id,
                ) = self.parser_one_line(line)

                candidate_news_index_batch.append(candidate_news_index)
                candidate_news_val_batch.append(candidate_news_val)
                click_news_index_batch.append(click_news_index)
                click_news_val_batch.append(click_news_val)
                candidate_news_entity_index_batch.append(candidate_news_entity_index)
                click_news_entity_index_batch.append(click_news_entity_index)
                label_list.append(label)
                impression_id_list.append(impression_id)
```
can be simply:
```
data = [] 
...
    data.append(self.parser_one_line(line))
```

Reply:
I'm not sure about this tbh, yes, it is cleaner, but at the same time the way it is now, it shows clearly all the components. This looks like a programming choice.

I think both options are equally valid, so I would ask @leavingseason if he wants to spend time in changing this

Reply:
yes I agree that that is a programming choice. Then I'm okay w/ the way it is now.